lab/src/Microsoft.ServiceModel.Syndication/src/ServiceDocumentFormatter.cs

```diff
@@ -122,180 +121,200 @@ protected static void LoadElementExtensions(XmlReader reader, CategoriesDocument
             if (collection == null)
             {
-                throw DiagnosticUtility.ExceptionUtility.ThrowHelperArgumentNull("collection");
+                throw new ArgumentNullException(nameof(collection));
             }
-            return collection.TryParseElement(reader, version);
+
+            return collection.TryParseElement(XmlReaderWrapper.CreateFromReader(reader), version);
         }
 
         protected static bool TryParseElement(XmlReader reader, ServiceDocument document, string version)
         {
             if (document == null)
             {
-                throw DiagnosticUtility.ExceptionUtility.ThrowHelperArgumentNull("document");
+                throw new ArgumentNullException(nameof(document));
             }
-            return document.TryParseElement(reader, version);
+
+            return document.TryParseElement(XmlReaderWrapper.CreateFromReader(reader), version);
         }
 
         protected static bool TryParseElement(XmlReader reader, Workspace workspace, string version)
         {
             if (workspace == null)
             {
-                throw DiagnosticUtility.ExceptionUtility.ThrowHelperArgumentNull("workspace");
+                throw new ArgumentNullException(nameof(workspace));
             }
-            return workspace.TryParseElement(reader, version);
+
+            return workspace.TryParseElement(XmlReaderWrapper.CreateFromReader(reader), version);
         }
 
         protected static bool TryParseElement(XmlReader reader, CategoriesDocument categories, string version)
         {
             if (categories == null)
             {
-                throw DiagnosticUtility.ExceptionUtility.ThrowHelperArgumentNull("categories");
+                throw new ArgumentNullException(nameof(categories));
             }
-            return categories.TryParseElement(reader, version);
+
+            return categories.TryParseElement(XmlReaderWrapper.CreateFromReader(reader), version);
         }
 
         protected static void WriteAttributeExtensions(XmlWriter writer, ServiceDocument document, string version)
         {
             if (document == null)
             {
-                throw DiagnosticUtility.ExceptionUtility.ThrowHelperArgumentNull("document");
+                throw new ArgumentNullException(nameof(document));
             }
+
             document.WriteAttributeExtensions(writer, version);
         }
 
         protected static void WriteAttributeExtensions(XmlWriter writer, Workspace workspace, string version)
         {
             if (workspace == null)
             {
-                throw DiagnosticUtility.ExceptionUtility.ThrowHelperArgumentNull("workspace");
+                throw new ArgumentNullException(nameof(workspace));
             }
+
             workspace.WriteAttributeExtensions(writer, version);
         }
 
         protected static void WriteAttributeExtensions(XmlWriter writer, ResourceCollectionInfo collection, string version)
         {
```

Comment:
It was being called by AtomPub10ServiceDocumentFormatter. Will Fix the method name and return a Task.

---
In reply to: [126762786](https://github.com/dotnet/wcf/pull/2028#discussion_r126762786) [](ancestors = 126762786)